holoviews/plotting/bokeh/callbacks.py

```diff
@@ -152,39 +164,84 @@ def __init__(self, plot, streams, source, **params):
         self.streams = streams
         self.comm = self._comm_type(plot, on_msg=self.on_msg)
         self.source = source
+        self.handle_ids = defaultdict(list)
 
 
     def initialize(self):
         plots = [self.plot]
         if self.plot.subplots:
             plots += list(self.plot.subplots.values())
 
```

Comment:
I'm still a bit confused about ``self.handle_ids`` -  I see it mentioned in four place:

1. When it is declared.
2. The bit in ``initialize`` above.
3. What looks like an a dictionary access in ``on_msg``.
4. The iteration over ``k, v`` you just mentioned.

In none of these places do I see ``self.handle_ids`` being changed after it is initialized. If you mean that ``self.handle_ids`` is being changed somewhere else in the code, then I wouldn't worry about using ``update`` here. In that case, I would also want to look at where the merging of ``self.handle_ids`` *is* happening to consider whether that should become part of the API of ``Callback``.